stripe/src/main/java/com/stripe/android/IssuingCardPinService.java

```diff
@@ -0,0 +1,274 @@
+                }
+            }
+        }
+        if (PIN_UPDATE.equals(action)) {
+            String cardId = (String) arguments.get("cardId");
+            String newPin = (String) arguments.get("newPin");
+            String verificationId = (String) arguments.get("verificationId");
+            String userOneTimeCode = (String) arguments.get("userOneTimeCode");
+            IssuingCardPinUpdateListener listener =
+                    (IssuingCardPinUpdateListener) arguments.get("listener");
+
+            try {
+                StripeApiHandler.updateIssuingCardPin(
+                        cardId,
+                        newPin,
+                        verificationId,
+                        userOneTimeCode,
+                        ephemeralKey.getSecret());
+
+                if (listener != null) {
+                    listener.onIssuingCardPinUpdated();
+                }
+            } catch (InvalidRequestException e) {
+                if (listener != null) {
+                    if ("expired".equals(e.getCode())) {
+                        listener.onError(
+                                CardPinActionError.ONE_TIME_CODE_EXPIRED,
+                                "The one-time code has expired",
+                                null);
+                    }
+                    if ("incorrect_code".equals(e.getCode())) {
+                        listener.onError(
+                                CardPinActionError.ONE_TIME_CODE_INCORRECT,
+                                "The one-time code was incorrect",
+                                null);
+                    }
+                    if ("too_many_attempts".equals(e.getCode())) {
+                        listener.onError(
+                                CardPinActionError.ONE_TIME_CODE_TOO_MANY_ATTEMPTS,
+                                "The verification challenge was attempted too many times",
+                                null);
+                    } else {
+                        listener.onError(
+                                CardPinActionError.UNKNOWN_ERROR,
+                                "An error occurred retrieving the PIN",
+                                e);
+                    }
+                }
+            } catch (APIConnectionException |
+                    APIException |
+                    AuthenticationException |
+                    CardException e) {
+                if (listener != null) {
+                    listener.onError(
+                            CardPinActionError.UNKNOWN_ERROR,
+                            "An error occurred retrieving the PIN",
+                            e);
+                }
+            }
+        }
+    }
+
+    @Override
+    public void onKeyError(int errorCode,
+                           @Nullable String errorMessage,
+                           @Nullable Map<String, Object> arguments) {
+        Object listener = arguments.get("listener");
+        if (listener == null){
```

Comment:
you can remove this check, instanceof covers null

Reply:
👍